src/vsr/replica.zig

```diff
@@ -4551,6 +4596,90 @@ pub fn ReplicaType(
             return self.commit_dispatch_resume();
         }
 
+        fn commit_stall(self: *Replica) enum { ready, pending } {
+            assert(self.commit_stage == .stall);
+            assert(!self.commit_stall_timeout.ticking);
+            assert(self.commit_prepare.?.header.op == self.commit_min + 1);
+
+            if (self.status == .normal and self.primary()) {
```

Comment:
I love explicit "functional" `if else` logic, but I also like early returns. In this case, when reading the code, I had to fold the if to confirm my suspicion that we do `return .ready;` in the else branch. So I'd probably early return here. 

Or maybe even do both, by splitting function into `commit_stall` and `commit_stall_primary`?

Doesn't matter, really!